src/blocks/external_ip.rs

```diff
@@ -0,0 +1,233 @@
+use std::thread;
+use std::time::Instant;
+
+use crossbeam_channel::Sender;
+use dbus::ffidisp::{BusType, Connection, ConnectionItem};
+use serde::{Deserialize as des, Serialize as ser};
+use serde_derive::Deserialize;
+
+use crate::blocks::{Block, ConfigBlock, Update};
+use crate::config::SharedConfig;
+use crate::errors::*;
+use crate::formatting::value::Value;
+use crate::formatting::FormatTemplate;
+use crate::http;
+use crate::scheduler::Task;
+use crate::util::country_flag_from_iso_code;
+use crate::widgets::text::TextWidget;
+use crate::widgets::{I3BarWidget, State};
+use crate::Duration;
+
+const API_ENDPOINT: &str = "https://ipapi.co/json/";
+const BLOCK_NAME: &str = "external_ip";
+
+#[derive(ser, des, Default)]
+struct IPAddressInfo {
+    #[serde(default = "bool::default")]
+    error: bool,
+    ip: String,
+    version: String,
+    city: String,
+    region: String,
+    region_code: String,
+    country: String,
+    country_name: String,
+    country_code: String,
+    country_code_iso3: String,
+    country_capital: String,
+    country_tld: String,
+    continent_code: String,
+    in_eu: bool,
+    #[serde(default)]
+    postal: String,
```

Comment:
`#[serde(default)]` doesn't help when postal in `null`.

Reply:
Fixed.